craft_application/pytest_plugin.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2025 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""A pytest plugin for assisting in testing apps that use craft-application."""
+
+from __future__ import annotations
+
+import os
+import pathlib
+import platform
+from collections.abc import Iterator
+from typing import TYPE_CHECKING
+
+import craft_platforms
+import pytest
+
+from craft_application import util
+from craft_application.util import platforms
+
+if TYPE_CHECKING:
+    from pyfakefs.fake_filesystem import FakeFilesystem
+
+
+@pytest.fixture(autouse=True, scope="session")
+def debug_mode() -> None:
+    """Ensure that the application is in debug mode, raising exceptions from run().
+
+    This fixture is automatically used. To disable debug mode for specific tests that
+    require it, use the :py:func:`production_mode` fixture.
```

Comment:
```suggestion
    This fixture is automatically used. To disable debug mode for specific tests,
    use the :py:func:`production_mode` fixture.
```
There is probably a better way to rephrase this, but the "it" here sounded like it was referring to debug mode, which made the sentence a bit confusing.